database/instance_db.go

```diff
@@ -72,6 +75,28 @@ func (d *LevelDBInstanceDB) Get(hash string) (*service.Instance, error) {
 	return i, tx.Commit()
 }
 
+// GetByService retrives instance by its service's hash.
+func (d *LevelDBInstanceDB) GetByService(hash string) (*service.Instance, error) {
```

Comment:
The relation between instance and service is many to one. many instance to one service.
This function should return an array of instance.

Reply:
to clarify: one service has many instances and one instance belongs to one service